kv/etcd/store.go

```diff
@@ -522,8 +522,13 @@ func newValue(val []byte, ver, rev int64) *value {
 	}
 }
 
-func (c *value) isNewer(other *value) bool {
-	return c.Rev > other.Rev
+func (c *value) IsNewer(other kv.Value) bool {
+	othervalue, ok := other.(*value)
```

Comment:
If so, I'd make it a type assertion so it'll panic if the value doesn't match what's expected instead of swallowing it silently